File: dart/scrabble-score/lib/scrabble_score.dart
const _scrabble = {
  [65, 69, 73, 79, 85, 76, 78, 82, 83, 84]: 1,
  [68, 71]: 2,
  [66, 67, 77, 80]: 3,
  [70, 72, 86, 87, 89]: 4,
  [75]: 5,
  [74, 88]: 8,
  [81, 90]: 10
};

num score(String word) => word
    .toUpperCase()
    .runes
    .map(valueForLetter)
    .fold(0, (total, value) => total + value);

num valueForLetter(int letter) =>
    _scrabble.entries.firstWhere((entry) => entry.key.contains(letter)).value;

File: kotlin/scrabble-score/src/main/kotlin/ScrabbleScore.kt
object ScrabbleScore {

    private val scrabble = mapOf(
        setOf('A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T') to 1,
        setOf('D', 'G') to 2,
        setOf('B', 'C', 'M', 'P') to 3,
        setOf('F', 'H', 'V', 'W', 'Y') to 4,
        setOf('K') to 5,
        setOf('J', 'X') to 8,
        setOf('Q', 'Z') to 10
    ).withDefault { 0 }

    fun scoreWord(word: String): Int =
        word.toUpperCase()
            .sumBy { scoreLetter(it) }

    private fun scoreLetter(letter: Char): Int =
        scrabble.keys
            .find { it.contains(letter) }
            .orEmpty()
            .let { scrabble.getValue(it) }
}
